app/services/participants/change_schedule/ecf.rb

```diff
@@ -159,6 +159,7 @@ def not_already_withdrawn
 
       def validate_new_schedule_valid_with_existing_declarations
         return if user.blank? || user_profile.blank?
+        return unless schedule
```

Comment:
I really don't like how those validations now rely on certain validations to have happened beforehand. 
I'd like to potentially experiment with the chain of responsibility pattern so chain validations. I don't really know if that would come out handy.